### PR TITLE
Integration with Commercetools

### PR DESCRIPTION
**Pull request by PikinerLiliya**
Changes:
- added integration with Commercetools;
- test coverage for new integration;
- updated readme file;
- changed initializing the SDK before the bundle loads, to avoid failure to connect.

### CODE REVIEW

**Comment by PikinerLiliya** [comment]
@pixelscript @petercoulton @JimAllanson Review it, please.

File: src/backends/CommerceTools.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      }
+      const headers = await this.getHeaders();
+      const idsStrings = '"' + filterIds.join('","') + '"';
+
+      const params = {
+        method: 'GET',
+        ...headers
+      };
+      const response = await fetch(`${this.apiUrl}/${projectKey}/product-projections?staged=false&limit=${PAGE_SIZE}&where=id in (${idsStrings})`, params);
+      const {results} = await response.json();
```
Comment (dev-warner) [inline src/backends/CommerceTools.js:54]: does the where need to be encoded?

File: src/backends/CommerceTools.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      console.error(e);
+      throw new ProductSelectorError('Could not get items', ProductSelectorError.codes.GET_SELECTED_ITEMS);
+    }
+  }
+
+  getImage(masterVariant) {
+    const largeImage = masterVariant.attributes && masterVariant.attributes.find(x => x.name === 'largeImageUrl');
+    return (largeImage && largeImage.value) || (masterVariant.images && masterVariant.images.length ? masterVariant.images[0].url : '')
+  }
+
```
Comment (dev-warner) [inline src/backends/CommerceTools.js:68]: I find the above a bit hard to read maybe refactor slightly if that's ok.

```javascript
function  getImage({ attributes = [], images = [] }) {
  const LARGE_IMAGE_KEY = 'largeImageUrl'
  const largeImage = attributes.find(({ name = '' }) => name === LARGE_IMAGE_KEY);
  
  if (largeImage && largeImage.value) {
    return largeImage.value;
  }
 
  return images.length ? images[0].url : '';
}
```

File: src/backends/CommerceTools.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    try {
+      const headers = await this.getHeaders();
+
+      const params = {
+        method: 'GET',
+        ...headers
+      };
+
+      const response = await fetch(`${this.apiUrl}/${projectKey}/product-projections/search?text.${this.locale}=${searchText}&staged=false&offset=${page.curPage * PAGE_SIZE}&limit=${PAGE_SIZE}`, params);
+      const {results, total} = await response.json();
```
Comment (dev-warner) [inline src/backends/CommerceTools.js:96]: might be a good idea to construct the url outside of the fetch and pass it in to make it a bit easier to read.

could maybe use new URL & .searchParam.append

File: public/index.html [hunk 4]
```diff
@@ -4,6 +4,26 @@
...
+          }
+    
+          script.src = 'https://unpkg.com/dc-extensions-sdk@1.0.3/dist/dc-extensions-sdk.umd.js';
+          document.head.appendChild(script);
+        });
+      }
+  
+      initialize();
+    </script>
+
```
[the remark this answers is not in the document]
Comment (PikinerLiliya) [reply public/index.html]: Done

File: src/backends/CommerceTools.js (hunk 1 above)
Comment (PikinerLiliya) [reply src/backends/CommerceTools.js:64]: Used qs.stringify

File: src/backends/CommerceTools.js (hunk 2 above)
Comment (PikinerLiliya) [reply src/backends/CommerceTools.js:84]: Done

File: src/backends/CommerceTools.js (hunk 3 above)
Comment (PikinerLiliya) [reply src/backends/CommerceTools.js:121]: Done

File: src/catalog-selector/CatalogSelector.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ export const CatalogSelectorComponent = params => {
         labelId="catalog"
         onChange={selectCatalog}
         labelWidth={labelWidth}
-        value={selectedCatalog}
+        value={selectedCatalog || ''}
```
[the remark this answers is not in the document]
Comment (PikinerLiliya) [reply src/catalog-selector/CatalogSelector.js:49]: Without it with latest node 
`Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
`

**Pull request merged**